views/components/Scroller.svelte

```diff
@@ -0,0 +1,195 @@
+<script context="module">
+  /*
+   * Adapted from https://github.com/sveltejs/svelte-scroller by Rich Harris, License: LIL
+   */
+
+  const handlers = [];
+  let manager;
+  if (typeof window !== "undefined") {
+    const run_all = () => handlers.forEach((fn) => fn());
+    window.addEventListener("scroll", run_all);
+    window.addEventListener("resize", run_all);
+  }
+  if (typeof IntersectionObserver !== "undefined") {
+    const map = new Map();
+    const observer = new IntersectionObserver(
+      (entries, observer) => {
+        entries.forEach((entry) => {
+          const update = map.get(entry.target);
+          const index = handlers.indexOf(update);
+          if (entry.isIntersecting) {
+            if (index === -1) handlers.push(update);
+          } else {
+            update();
+            if (index !== -1) handlers.splice(index, 1);
+          }
+        });
+      },
+      {
+        rootMargin: "400px 0px", // TODO why 400?
```

Comment:
Remove or elaborate comment

Reply:
I removed the comment in https://github.com/nzzdev/Q-scroll-graphic/pull/1/commits/75e61359f0be5faaf9854c4dcf58b288c76381a9. This is actually coming from the source repository [svelte-scroller](https://github.com/sveltejs/svelte-scroller)